File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp={}
        ans=[]
        for num in nums:
            if num in mp:
                mp[num]+=1
            else:
                mp[num]=1
        
        arr = sorted(mp.items(), key=lambda x: x[1], reverse=True)
        for i in range(k):
            ans.append(arr[i][0])

        return ans

        

File: Data Structures & Algorithms/longest-turbulent-subarray/submission-1.cpp
class Solution {
public:
    int maxTurbulenceSize(vector<int>& arr) {
        int n = arr.size();
        if (n == 1) return 1;

        int inc = 1, dec = 1,ans=1;
        for(int i=0;i<n-1;i++){
            if(arr[i]<arr[i+1]){
                inc=dec+1;
                dec=1;
            }else if(arr[i]>arr[i+1]){
                dec=inc+1;
                inc=1;
            }else{
                dec=1,inc=1;
            }
            ans=max(ans,max(inc,dec));
        }
        return ans;
        
    }
};